MIGRATION.md

```diff
@@ -53,13 +60,59 @@ Tip: Provide the `--transform`/`-t` argument at the end of the command, so that
+
+### Accessibility
+
+Any accessible label props that were previously optional are now required and enforced in all components.
+
+To help identify places where accessible labels are missing (even if an app is built with TypeScript, some configurations will not break the build if required props are missing in JavaScript files), components will throw runtime errors in development at missing labels. Production and testing builds are not affected.
+
+During the migration and while the missing labels are still being added, you can use an escape hatch to continue running the app in development without throwing accessibility errors.
+
+In your app, expose the `UNSAFE_DISABLE_ACCESSIBILITY_ERRORS` environment variable. You can use the [Webpack `DefinePlugin`](https://webpack.js.org/plugins/define-plugin/) ([here's an example](https://github.com/sumup-oss/circuit-ui/blob/main/.storybook/main.js#L45-L53) in the Circuit UI Storybook config) or, if your app uses Next.js, you can declare the variable in your `next.config.js` ([here's an example](https://github.com/sumup/ze-dashboard/blob/master/next.config.js#L78) in the SumUp merchant dashboard).
+
+Now, if you want to turn off the accessibility errors temporarily, run the development app with the environment variable set to `true`:
+
+```sh
+UNSAFE_DISABLE_ACCESSIBILITY_ERRORS=true yarn dev # or yarn start
+```
+
+Keep in mind that this escape hatch is not meant as a way to permanently avoid the errors, but as a temporary workaround while the missing labels are being written, localized and added to the relevant components.
+
+> Note: for the `Input` and `Select` components, use the built-in `label` prop instead of using the `Label` component separately.
+
+Other accessibility improvements include:
+
+- Accessible modals and popovers, see [the dedicated section](#consolidated-modals-and-popovers)
+- Semantic heading elements, see [the typography section](#heading-as-props)
+
+### New JSX transform
+
+Circuit v3 improves compatibility with the [new JSX transfors](https://reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html) (introduced inReact v17).
+
+Applications on Next.js and Emotion 10 (Emotion 11 is not yet supported by Circuit UI) can now use the following Babel config:
 
-### JSX runtime
+```json
+{
+  "presets": [
+    [
+      "next/babel",
+      {
+        "preset-react": {
+          "runtime": "automatic",
+          "importSource": "@emotion/core"
+        }
+      }
+    ]
+  ],
+  "plugins": [["babel-plugin-emotion", { "cssPropOptimization": true }]]
+}
+```
 
-(...)
+See the [Emotion 10 docs](https://5faaafd0bd0f3f0008469537--emotion.netlify.app/docs/css-prop#babel-preset) for more details on the required config.
```

Comment:
Is this required to use Circuit UI or just to use Emotion with the `css` prop in the application?

I'd assume that it's not required for Circuit UI since it's pre-transpiled with these settings.

Reply:
Ah, maybe that's only to use Emotion with `css`. Would you recommend simplifying the example config?